src/light-clients/ethereum.md

```diff
@@ -0,0 +1,143 @@
+BLS signatures (Boneh-Lynn-Shacham) are a type of cryptographic signature scheme that allows multiple signatures to be aggregated into a single signature. This makes them space and compute efficient (you can aggregate hundreds of signatures into one). Just as we aggregate signatures, we can aggregate public keys as well, such that the aggregate public key can verify the aggregated signature.
+
+For our SyncAggregate, computing the aggregate pubkey is simple:
+
+```python
+def _aggregate_pubkeys(committee, bits)
+    pubkeys = []
+    for i, bit in enumerate(bits):
+        if bit:
+            pubkeys.append(committee[i])
+    return bls.Aggregate(pubkeys)
+```
+
+At scale, we can aggregate thousands (if not hundreds of thousands) of signatures and public keys, whily only verifying their aggregates.
+
+To our light client, as long as a majority of sync committee members have attested the block, it is consider final.
+
+```python
+class LightClient():
+    def update(self, block: BeaconBlockBody):
+
+        # Count how many committee members signed
+        signature_count = sum(sync_aggregate.sync_committee_bits)
+
+        # Need 2/3+ committee participation for finality
+        if signature_count < (SYNC_COMMITTEE_SIZE * 2) // 3:
+            raise ValueError("Insufficient signatures from sync committee")
+
+        # Construct aggregate public key from the current committee and bit vector
+        aggregate_pubkey = _aggregate_pubkeys(
+            self.current_sync_committee,
+            block.sync_aggregate.sync_committee_bits
+        )
+```
+
+Now we have the `aggregate_pubkey` for the committee, as well as verified that enough members have signed. Notice that to obtain the sync committee public keys, we used `self.current_sync_committee`. This is set during initialization, and later updated in our `update` function.
+
+Next we have to construct the digest (what has been signed) before we verify the aggregated signature. If we didn't compute the digest ourselfs, but obtained it from the block, then the caller could fraudelenty pass a correct digest, but have other values in the block altered.
+
+```python
+    signing_root = self._compute_signing_root(block)
+
+    # Verify the aggregated signature against the aggregated public key
+    if not bls.Verify(
+        aggregate_pubkey,
+        signing_root,
+        sync_aggregate.sync_committee_signature
+    ):
+        raise ValueError("Invalid sync committee signature")
+```
+
+Since the signature and block are both valid, we can now trust the contents of the passed beaconblock. Next the lightclient will store data from the block
+
+```python
+    self.latest_block_root = self._compute_block_root(block)
+    self.latest_slot = block.slot
+```
+
+Finally, we have to update the sync committee. The committee rotates every epoch, and thus if this is at the boundary, we have to update these values. Luckily Ethereum makes this easy for us, and provides what the next sync committee will be:
```

Comment:
Not every epoch but on every sync committee period. There are for example 512 or 256 (i dont remember the exact number) epochs in every sync committee period. Epochs are used for reaching finality.